### PR TITLE
Split container - Remove dependencies on split.js

### PR DESCRIPTION
**Pull request by deltakosh**
cc @AmoebaChant to do it on Smart Filters editor too

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15852/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15852/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15852/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Review by sebavan — APPROVED** [review]
love it @PatrickRyanMS can you test on the preview ?

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15852/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15852/merge/testResults/webgl2playwright/index.html

File: packages/dev/sharedUiComponents/src/split/splitter.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            elementRef.current.style.height = `100%`;
+            elementRef.current.classList.add(styles["horizontal"]);
+        } else {
+            elementRef.current.style.height = `${props.size}px`;
+            elementRef.current.style.width = `100%`;
+            elementRef.current.classList.add(styles["vertical"]);
+        }
+
+        splitContext.init(elementRef.current, props.controlledSide, props.initialSize, props.minSize, props.maxSize);
+    });
```
Comment (RaananW) [inline packages/dev/sharedUiComponents/src/split/splitter.tsx:73]: I assume you mean to add `[]` here as a 2nd parameter? or do you want it to run on every re-render?
```suggestion
    }, []);
```

Comment (deltakosh) [reply packages/dev/sharedUiComponents/src/split/splitter.tsx:73]: I need it on every rerender else I will lose the context on the split container (which stores the various arrays). I tried with [] but then if the control is regenerated the arrays (like isfloating) are empty

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Pull request merged**